nssp_demo/collate_score_tables.R

```diff
@@ -0,0 +1,212 @@
+library(data.table)
+library(argparser)
+
+#' Get all the subdirectories within a parent directory
+#' that match the pattern for a forecast run for a
+#' given disease and optionally a given report date.
+#'
+#' @param parent_dir Directory in which to look for forecast subdirectories.
+#' @param diseases Names of the diseases to match, as a vector of strings,
+#' or a single disease as a string.
+#' @return A vector of paths to the forecast subdirectories.
+get_all_forecast_dirs <- function(dir_of_forecast_date_dirs,
+                                  diseases) {
+  # disease names are lowercase by convention
+  match_patterns <- stringr::str_c(tolower(diseases), "_r")
+  matches_diseases <- \(x) {
+    any(stringr::str_starts(x, pattern = match_patterns))
+  }
+
+  dirs <- tibble::tibble(
+    dir_path =
+      fs::dir_ls(dir_of_forecast_date_dirs,
+        type = "directory"
+      )
+  ) |>
+    dplyr::mutate(dir_name = fs::path_file(dir_path)) |>
+    dplyr::filter(purrr::map_lgl(dir_name, matches_diseases)) |>
+    dplyr::pull(dir_path)
+
+  return(dirs)
+}
+
+process_loc_date_score_table <- function(model_run_dir) {
+  table_path <- fs::path(model_run_dir,
+    "score_table",
+    ext = "rds"
+  )
+  location <- fs::path_file(model_run_dir)
+  if (!(fs::file_exists(table_path))) {
+    warning(glue::glue(
+      "No `score_table.rds` found for location ",
+      "{location} in directory {model_run_dir}"
+    ))
+    return(NULL)
+  }
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$location <- location
+  score_table$sample_scores$location <- location
+  return(score_table)
+}
+
+process_date_score_table <- function(date_fit_dir) {
+  table_path <- fs::path(date_fit_dir,
+    "score_table",
+    ext = "rds"
+  )
+  table_dir <- fs::path_file(date_fit_dir)
+
+  report_date <- stringr::str_match(
+    table_dir,
+    "r_(([0-9]|-)+)_f"
+  )[2] |>
+    lubridate::ymd()
+
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$report_date <- report_date
+  score_table$sample_scores$report_date <- report_date
+  return(score_table)
+}
+
+bind_tables <- function(list_of_table_pairs) {
+  sample_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$sample_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+  quantile_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$quantile_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+
+  sample_scores <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      x$sample_scores
+    }
+  ) |>
```

Comment:
Equivalently and more compactly:
```suggestion
    list_of_table_pairs, "sample_scores") |>
```